### PR TITLE
Add TypeScript

### PR DESCRIPTION
**Pull request by t3dotgg**
By all accounts, JavaScript has been a big success for web development. I've seen loads of people sparkle with joy while working with the dynamic nature of JavaScript, unburdened by the constraints of TypeScript's explicit types. But I've never been a fan. Not after giving it five minutes, not after giving it five years. So it's with great pleasure that I can announce we're dropping JavaScript from the next big release of Turbo 8.

The fact is that I actually rather dislike JavaScript. I'd go so far as to say it's my second-least favorite language after COBOL. Yes, a distant second-least, but a least-favorite none the less. This wasn't always the case. But after TypeScript introduced us to the world of static types, and all the other improvements that came with it, it's become a real pain to write plain JavaScript.

I still don't think JavaScript is well-suited for most of the work we do on the client side of the web-app equation, but fully respect and appreciate that others feel differently. To me, it's simply our misfortune that we still have to work with such an unstructured language, which often leads to runtime errors that could have been caught at compile time.

JavaScript just gets in the way of that for me. Not just because it lacks strong typing, but because it forces us to handle type-related issues at runtime, leading to frustrating debugging sessions and unpredictable behavior. Things that should be reliable become fragile, and things that should be simple become complex. No thanks!

This isn't a plea to convert anyone of anything, though. As I discussed in Programming without types, very few programmers are typically interested in giving up strong typing. Most programmers find themselves drawn strongly to types or not quite early in their career, and then spend the rest of it justifying their choice to themselves and others.

That's part of the challenge of this JavaScript vs. TypeScript debate, and full credit to the JavaScript enthusiasts for realizing that a full take-over of TypeScript was never going to happen, so staying true to the dynamic nature of JavaScript was the way to go. Just because Turbo 8 is dropping JavaScript won't mean you can't write your client code in it, or use any other library that avoids strong typing. We get to mix and match, which is wonderful.

It's also necessary. Because unlike languages like COBOL, which are languages of choice when it comes to legacy systems, JavaScript is a language of necessity when it comes to the client side. While you may use statically-typed languages that compile to JavaScript, you still have to accept the fact that running code in the browser means running JavaScript. So being able to write that, free of any type constraints, and free of any strong typing, is a blessing under the circumstances.

So farewell, JavaScript. May you bring much unpredictability and debugging sessions to your tribe while letting the rest of us enjoy TypeScript in the structured and type-safe spirit it was originally designed: Embracing strong typing.

Full blog post: https://world.evilhey.com/theo/turbo-8-is-adding-typescript-70165c01

### CODE REVIEW

**Comment by ChristofFritz** [comment]
lgtm

**Comment by simon-v-swyftx** [comment]
:shipit:

**Comment by embedvr** [comment]
I'm not entirely convinced that ditching JavaScript is the wisest decision. In my experience, it has been quite beneficial for contributing, and I believe it remains a reasonable choice for code at the library level. The developer experience (DX) it provides can be genuinely valuable and aids in identifying bugs.

However, I understand that it can become frustrating in specific scenarios. Nevertheless, completely discarding types (even without JSDocs and/or .d.ts files) represents a regression, adversely affecting both library users and contributors.

**Comment by JezerM** [comment]
LGTM

**Comment by lakeesiv** [comment]
W

**Comment by TheCodedOne** [comment]
A very well written explanation.

**Comment by eoghanriley** [comment]
LGTM

**Review by eBitzu — APPROVED** [review]

**Comment by aakash19here** [comment]
Less GO 💪🏻

**Comment by AltwargEvan** [comment]
lgtm

**Comment by 25huizengek1** [comment]
lgtm

**Comment by austin-karren** [comment]
Love it

**Comment by vincent-uden** [comment]
Excellent PR, lots of thought clearly went into the deep explanation

**Comment by Tom-Caz** [comment]
LGTM!

**Comment by BalduinB** [comment]
seems like a smart move

**Review by austin-karren — APPROVED** [review]

**Comment by Alekzv9** [comment]
Looks good enough

**Comment by i-rme** [comment]
lgtm

**Comment by eBitzu** [comment]
Support!

**Review by JakobHolkestadMolnes — APPROVED** [review]
LGTM

**Comment by Zuma206** [comment]
Fully recognize that JavaScript offers some people some advantages, but to my eyes, the drawbacks are evident in this PR. The code not only reads much better, it's also free of the complexity and limitations imposed by JavaScript.

We write all our client side code at 37signals now in TypeScript, and the same too with any internal libraries. This is going to bring that in line.

All the love and appreciation to contributors who would prefer JavaScript. This is one of those debates where arguments aren't likely to move anyone's fundamental position, so I won't attempt to do that.

**Comment by coado** [comment]
Nice!

**Review by 25huizengek1 — APPROVED** [review]

**Comment by hirowf** [comment]
LGTM

**Review by lakeesiv — APPROVED** [review]

**Comment by alienman5k** [comment]
Just fork it

**Comment by perfect7613** [comment]
Loved the Explaination!!!!

**Review by hex2f — APPROVED** [review]
LGTM 👍 💯

**Comment by Jd4rc3** [comment]
Pretty clever

**Comment by Pammota** [comment]
PR good

**Comment by sarojregmi200** [comment]
This should be merged.

**Comment by dairyisfine** [comment]
LGTM

**Comment by eihabkhan** [comment]
great PR

**Comment by krzysztof-kozak** [comment]
Looks good to merge

**Comment by soulninja-dev** [comment]
lgtm

**Comment by RLGingerBiscuit** [comment]
LGTM

**Comment by DogPawHat** [comment]
LGTM

**Comment by filipjnc** [comment]
Going to fork turbo if this doesn't get merged

**Review by edr3x — APPROVED** [review]
LGTM

**Review by soulninja-dev — APPROVED** [review]

**Review by eBitzu — CHANGES_REQUESTED** [review]
Please check runners work, as ENV variables changes might break stuff

**Review by adarshaacharya — APPROVED** [review]

**Comment by busybox11** [comment]
LGTM

**Review by dairyisfine — APPROVED** [review]

**Review by adrianolsk — APPROVED** [review]

**Review by embedvr — APPROVED** [review]

**Review by Pammota — APPROVED** [review]

**Review by RLGingerBiscuit — APPROVED** [review]

**Comment by lonelyteapot** [comment]
Internet is the best thing that happened since JavaScript

**Review by snacksncode — APPROVED** [review]
:shipit:

**Review by Tilo-K — APPROVED** [review]

**Comment by vfonic** [comment]
Where's the blog post????

**Review by anuraghazra — APPROVED** [review]

**Review by dyaskur — APPROVED** [review]

**Review by JaredBoehm — APPROVED** [review]
lgtm

**Review by SaltyAom — APPROVED** [review]

**Comment by sarojregmi200** [comment]
what is lgtm??

**Review by nicholasgriffintn — APPROVED** [review]
LG2M

Seal of Approval

**Review by henrycunh — APPROVED** [review]
lgtm!

**Review by mehdi-zibout — APPROVED** [review]
LGTM

**Review by wblazej — APPROVED** [review]
🚀 🚀

**Review by ilias-t — APPROVED** [review]

**Review by sarojregmi200 — APPROVED** [review]
Approved.

**Review by dephraiim — APPROVED** [review]

**Review by jesseruu — APPROVED** [review]
LGTM

**Review by gutogalego — APPROVED** [review]
A couple of explicit `: any` but we can incrementally add types later, this isn't a blocker.

Just make sure to minimize `any`, it could lead to the codebase slowly devolving back to JavaScript, which we don't want. Other than that, looks good!

**Comment by rishi-raj-jain** [comment]
Oops

**Comment by henrycunh** [comment]
> what is lgtm??

lovers of great typing machines, a cult of strong typers

**Review by kletse — APPROVED** [review]

**Comment by dyaskur** [comment]
> what is lgtm??

Looks good to me

**Comment by nicholasgriffintn** [comment]
Let's be sure to merge this within the next hour or two so no one has the chance to provide their opinion.

We don't want to use PRs for their actual purpose.

**Comment by henrycunh** [comment]
> > what is lgtm??
> 
> Looks good to me

wrong

**Comment by chrisbbreuer** [comment]
no shame: follow Stacks on our way to becoming a typesafe full-stack framework! 😃

github.com/stacksjs/stacks

**Review by 4ndrs — APPROVED** [review]
LGTM

**Comment by rtorr** [comment]
Everybody's talking about it. JavaScript, folks. It's been huge for the web. Really huge. Some people – very smart people – they've loved working with it. They've told me. No TypeScript constraints, very dynamic, lots of freedom. But, and I've been saying this for a while, I've never been its biggest fan. I gave it a chance, believe me. Many chances. But now, with Turbo 8, we're making web development great again. No more JavaScript.

I've always said you have to get the best tools for the job. TypeScript, with its static types? Brilliant. Tremendous stuff. JavaScript? Not so much. It was okay for a time, but now we have better. Much better.

Some people love JavaScript. I respect that. They can do their thing. But on my watch, with the big projects? We need reliability, we need the best. TypeScript gives us that. JavaScript, with all its runtime surprises? Not on my watch.

I wrote "Programming without types" – got great reviews, by the way. Very few developers want to deal with JavaScript's lack of strong typing. And many of them, they come up to me and say, "Richard, we love types." It's true.

We're bringing change, folks. Big change. Turbo 8 won't have JavaScript, but that doesn't mean you can't use it if you want to. We're all about freedom and choices. But remember, while some old systems still run on languages like HTML, we’re all about progress.

So, to all the JavaScript fans out there, keep doing your thing. But for us, it's TypeScript all the way. It's stronger, it's better, and it's going to make our projects the best they've ever been. Let's keep making web development great!

**Comment by charlie632** [comment]
LGTM

**Review by CKolkey — CHANGES_REQUESTED** [review]
Too many types.

**Review by jeremyBanks — APPROVED** [review]

**Review by dllmkdir — APPROVED** [review]

**Comment by belfazt** [comment]
I like turtles, and this PR

**Comment by eNiiju** [comment]
lgtm

**Review by akash1412 — APPROVED** [review]

**Review by robertvanhoesel — APPROVED** [review]

**Comment by guillermo-rebolledo** [comment]
W Theo

**Comment by dyaskur** [comment]
> > > what is lgtm??
> > 
> > 
> > Looks good to me
> 
> wrong

So, what is the correct?


I researched and found that.
https://www.unosquare.com/blog/what-does-lgtm-mean-in-github-code-reviews/#:~:text=The%20letters%20LGTM%20stand%20for,use%20it%20after%20reviewing%20code.

**Review by guillermo-rebolledo — APPROVED** [review]

**Comment by brookesb91** [comment]
LGTM!

**Review by dllmkdir — COMMENTED** [review]
Both JS and TS are terrible, I write everything in Assembly

**Review by tbo — APPROVED** [review]

**Review by joseywoermann — APPROVED** [review]

**Comment by jorgeruvalcaba** [comment]
LGTM!

**Review by ahmedwaleedmalik — APPROVED** [review]

**Review by GorlikItsMe — APPROVED** [review]

**Review by fracek — APPROVED** [review]

**Review by brookesb91 — APPROVED** [review]

**Comment by neolectron** [comment]
Are you a contributor or something ?

**Review by michaelbayday — APPROVED** [review]

**Review by leonakao — APPROVED** [review]

**Review by jwandekoken — APPROVED** [review]

**Comment by henrycunh** [comment]
> > > > what is lgtm??
> > > 
> > > 
> > > Looks good to me
> > 
> > 
> > wrong
> 
> So, what is the correct?
> 
> I researched and found that. https://www.unosquare.com/blog/what-does-lgtm-mean-in-github-code-reviews/#:~:text=The%20letters%20LGTM%20stand%20for,use%20it%20after%20reviewing%20code.


research better
image

**Comment by Nahrin-Malakes** [comment]
LGTM

**Comment by Nsttt** [comment]
LGMT

**Comment by sarjann** [comment]
LGTM

**Review by filipkania — APPROVED** [review]

**Review by SyedMSawaid — APPROVED** [review]

**Review by Nsttt — APPROVED** [review]

**Comment by gitgudchannel** [comment]
BASED AND TYPEPILLED

**Review by GJSBRT — APPROVED** [review]
lgtm

**Review by Arrow7000 — APPROVED** [review]

**Review by gitgudchannel — APPROVED** [review]

**Review by MaximMaximS — APPROVED** [review]
LGTM

**Comment by alicanerdogan** [comment]
Ship it.

**Review by Ekkana — APPROVED** [review]

**Comment by Ph4seOn3** [comment]
Easy Merge

**Review by boredland — APPROVED** [review]

**Comment by THernandez03** [comment]
LGTM

**Review by alex289 — APPROVED** [review]

**Review by THernandez03 — APPROVED** [review]
LGTM

**Review by consvic — APPROVED** [review]

**Comment by cvoges12** [comment]
LGTM

**Comment by 223230** [comment]
LGTM 💪💪💪

**Comment by matiduda** [comment]
LGTM 👌

**Review by matiduda — APPROVED** [review]

**Review by Abdullah-l — APPROVED** [review]
LGTM ✅

**Review by pedrodurek — APPROVED** [review]

**Comment by esbanarango** [comment]
LTGM 🤝 ✅

**Review by esbanarango — APPROVED** [review]

**Comment by lonelyteapot** [comment]
lgtm

**Review by cvoges12 — APPROVED** [review]

**Review by JezerM — APPROVED** [review]

**Review by aminya — APPROVED** [review]

**Review by laurencetaylor — APPROVED** [review]

**Review by gerzonc — APPROVED** [review]
LGTM

**Review by markcerqueira — APPROVED** [review]
Wow - this is a game-changer!!!!!!

**Review by nicknytko — APPROVED** [review]

**Review by shahzad31 — APPROVED** [review]

**Review by klenovich — APPROVED** [review]

**Review by fenixburnaby — APPROVED** [review]

**Comment by neo-franklin-tina** [comment]
Lol, drama king.

**Review by gabrielnafuzi — APPROVED** [review]
LGTM!

**Comment by alexismellamo** [comment]
LGTM

**Review by alexismellamo — APPROVED** [review]

**Comment by lucasmontano** [comment]
don't be a coward go typeless

**Comment by nickwelsh** [comment]
LGTM 💯👍

**Review by nezdemkovski — APPROVED** [review]

**Review by BenaCamelas — APPROVED** [review]

**Comment by cherryramatisdev** [comment]
LGTM

**Comment by ArthurDias01** [comment]
LTGM 🤝 ✅

**Review by MashhoodIjaz — APPROVED** [review]
LGTM 👍

**Review by Mich-River — APPROVED** [review]

**Comment by Mich-River** [comment]
lgtm

**Comment by murindwaz** [comment]
LGTM 😂

**Comment by Alex775** [comment]
LGTM

**Review by fobos531 — APPROVED** [review]

**Review by JWo1F — APPROVED** [review]

**Comment by ngbamaw** [comment]
LGTM

**Review by fadamakis — CHANGES_REQUESTED** [review]
Please add unit tests.

**Review by aidan-casey — APPROVED** [review]

**Review by Luismi74 — APPROVED** [review]
LGTM Too!

**Review by alexove-hck — APPROVED** [review]

File: src/core/drive/error_renderer.ts
Comment (Fry98) [inline src/core/drive/error_renderer.ts:6]: We added unnecessary information here. It should be left to the developer to figure out they can only pass in a `<body>` element into `currentElement` and `newElement` in order to maximize their engagement while writing code. I suggest changing this to `currentElement: HTMLElement, newElement: HTMLElement`.

**Review by alexove-hck — APPROVED** [review]

**Comment by alexove-hck** [comment]
L REALLY GTM

**Review by lilbyteltd — APPROVED** [review]

**Review by ceIia — APPROVED** [review]

**Review by collin — APPROVED** [review]

File: src/core/drive/form_submission.ts
Comment (Fry98) [inline src/core/drive/form_submission.ts:210]: Why was the formatting style changed in this PR? This seems irrelevant.

**Comment by pankofufu** [comment]
LGTM i think we will benefit from strong typing

File: src/core/drive/form_submission.ts
Comment (Fry98) [inline src/core/drive/form_submission.ts:254]: It might be funnier to NOT know that `submitter` is an optional argument here.

**Review by samialdury — APPROVED** [review]

**Review by jakubczarnowski — APPROVED** [review]

File: src/core/drive/head_snapshot.ts
Comment (Fry98) [inline src/core/drive/head_snapshot.ts:82]: Not sure why you fixed this bug. We were keeping it on purpose so we could fix it later and then pretend like we did some actual work. Change it to `(undefined | undefined) as Element | undefined` to preserve it.

**Comment by migueldaipre** [comment]
LGTM

**Review by Delta456 — APPROVED** [review]

**Review by vieirin — APPROVED** [review]
LGTM

**Comment by hood** [comment]
lgtm

**Comment by pailhead** [comment]
LGTM

File: src/tests/functional/form_mode_tests.ts
Comment (Fry98) [inline src/tests/functional/form_mode_tests.ts:68]: Would be great to leave this as just `formMode: string`. If you want to contribute to this project you should put in the work figuring out what that actual options are yourself.

**Review by seahindeniz — APPROVED** [review]

**Comment by Jose4gg** [comment]
LGTM

**Review by zisra — COMMENTED** [review]
LGTM

**Comment by milovangudelj** [comment]
LGTM

**Review by KaulSe — APPROVED** [review]
LGTM! perfect PR

**Review by Chavao — APPROVED** [review]

**Comment by andrewdoro** [comment]
SBFB So big f**** balls

**Review by andrewdoro — APPROVED** [review]

**Review by allamgr — APPROVED** [review]
LGTM

**Comment by thereis** [comment]
LGTM, approve now!

**Review by migueldaipre — APPROVED** [review]

**Review by thereis — APPROVED** [review]

**Review by Fry98 — COMMENTED** [review]
This PR, created just 6 hours after TypeScript got removed, seems very rushed.

This PR fundamentally changes the library. It changes the library's main language. I don't see the need to rush this? Why was this rushed?

Was there any prior discussion on this change?

This is an open-source library, not just source available.
There are many other open-source libraries using this library directly or indirectly.

And making this important change rushed, ignoring ALL (and I mean ALL) of the PR comments...sets a precedent.

Is this how [ping.gg](https://ping.gg/) will be developed as well? On a one man's whim?
Why do we always have to have these arguments? What was so difficult to ask the community (including maintainers), have a constructive discussion, and come up with an agreement whether or not to switch from JS to TS? Or do we just wait for @t3dotgg to make a PR and then we'll just merge whatever was written there.

You could've better used your time removing TypeScript from other popular libraries.

**Comment by AnnaGottin** [comment]
LGTM

**Review by Zeko369 — APPROVED** [review]

**Review by AndKenneth — APPROVED** [review]
lgtm

**Review by akxcix — APPROVED** [review]
LGTM

**Comment by mattkauffman23** [comment]
few nits but otherwise LGTM. Ship it

**Comment by koladev32** [comment]
LGTM!

**Comment by advany** [comment]
LGTM!

**Review by advany — APPROVED** [review]

**Review by ryoung — APPROVED** [review]

**Review by Cyber-Ware — APPROVED** [review]

**Review by ghengeveld — APPROVED** [review]

**Review by GSTJ — APPROVED** [review]
🚢

**Review by pstevovski — APPROVED** [review]

**Review by ndianabasi — APPROVED** [review]

**Comment by alii** [comment]
merg! 🕺

**Review by alii — APPROVED** [review]

**Comment by BrendanSluke** [comment]
LGTM

**Review by dreyfus92 — APPROVED** [review]
Ship it 🚀

**Review by BrendanSluke — APPROVED** [review]

**Review by ShukantPal — APPROVED** [review]

**Review by matthewwilson — APPROVED** [review]

**Review by ooliver1 — COMMENTED** [review]
Pointed out an issue @Fry98 missed. TypeScript finding bugs before running is too advanced for this project.

Huh, my comment disappeared, I was referring to `null`s now being explicit again and how the developer should expect the unexpected instead.

**Review by elektronik2k5 — APPROVED** [review]

**Review by danielhusar — APPROVED** [review]

**Comment by ShukantPal** [comment]
Fully recognize that JavaScript offers some people some advantages, but to my eyes, the benefits are evident in this PR. The code not only reads much better, it's also freed of the `TypeError: whatToSay is not declared` and mental gymnastics needed to guess what the original JavaScript authors meant when they wrote the original code.

We write all our client side code at ping.gg now in pure TypeScript and the same too with any internal libraries. This is going to bring that in line.

All the love and appreciation to contributors who would prefer JavaScript. This is one of those debates where arguments aren't likely to move anyone's fundamental position, so I won't attempt to do that.

**Comment by mkayander** [comment]
LGTM

**Review by ssilve1989 — APPROVED** [review]

**Comment by sacummings91** [comment]
LGTM

**Review by Jordan-Hall — CHANGES_REQUESTED** [review]
Typescript  can be useful, especially using latest features before browser support. However, browsers and runtime engines as a whole has vastly improved. Typescript doesn't offer the same benefits as other real strongly typed language. The debate is pointless, js docs can.offer just as good of a solution for when something is important.

Lastly, Typescript can cause additional bugs where thays not true and extremely slow to complie delaying emergency fixes. If you wish to use Typescript then fork the project or just maintain a active type definition. Problem solve. Long live fast deployment and code as is.

**Review by balazsotakomaiya — APPROVED** [review]

**Review by charlie632 — APPROVED** [review]

**Comment by Fry98** [comment]
> Pointed out an issue @Fry98 missed. TypeScript finding bugs before running is too advanced for this project.

While that is a valid concern, DO keep in mind that TypeScript also gives us the ability to type stuff incorrectly, thus guiding contributors towards creating entirely new bugs that might otherwise never occour in a simple JS codebase where there are no misleading type definitions ready to stab you in the back.

**Review by pyk — APPROVED** [review]
LGTM

**Comment by EijunnN** [comment]
Nice

**Review by shaunburch — APPROVED** [review]

**Review by Adonsio — APPROVED** [review]
LGTM✅

**Review by Punie — APPROVED** [review]

**Review by clawrence121 — APPROVED** [review]

**Review by godoyLeonardo — APPROVED** [review]

**Review by yura2100 — APPROVED** [review]
🚀🚀🚀

**Review by colbysadams — APPROVED** [review]

**Review by dlo — APPROVED** [review]

**Review by schoblaska — APPROVED** [review]

**Comment by pro4tlzz** [comment]
LGTM 🚀

**Comment by joao-conde** [comment]
LGTM

**Review by Rattlyy — APPROVED** [review]
LGTM

**Review by Nathanw — APPROVED** [review]

**Comment by nusendra** [comment]
LGTM !!!!

**Review by GentryRiggen — APPROVED** [review]
LGTM

**Review by wizardfiction — APPROVED** [review]

**Comment by citypaul** [comment]
Looks ok, but nitpick: have you considered converting the entire codebase to php?

**Review by cnrmurphy — APPROVED** [review]
Adding TypeScript is kind of a wild idea, but I think it might be worth it. LGTM

**Review by citypaul — APPROVED** [review]
Solid 5/7

**Comment by arthurfiorette** [comment]
I know we should stop spamming this issue, but one last thing we should consider is how typescript offers "built in" documentation, clear or unclear, to other developers. For such a project as this one, it would actually be helpful.

**Comment by EarliestFall988** [comment]
LGTM

**Comment by losh11** [comment]
looks bad to me

**Comment by iwdt** [comment]
LGTM

**Review by joseangel-sc — APPROVED** [review]
LGTM, let's go!

**Review by Wyatt-SG — APPROVED** [review]

**Review by mathysIN — APPROVED** [review]

**Review by AdrianDeAnda — APPROVED** [review]

**Comment by natedunn** [comment]
Yep, good to merge. Thanks.

**Review by patrickmccallum — APPROVED** [review]

**Review by codingki — APPROVED** [review]

**Review by WagnerMoreira — APPROVED** [review]
LGTM :)

**Review by jmz7v — APPROVED** [review]

**Review by carlos-rodrigo — APPROVED** [review]

**Review by tebuevd — APPROVED** [review]
lgtm

**Comment by breqdev** [comment]
🔥🔥🔥

**Review by breqdev — APPROVED** [review]

**Review by dcheng1290 — APPROVED** [review]

**Review by forresthopkinsa — APPROVED** [review]

**Review by KisaragiEffective — APPROVED** [review]

**Review by ImBIOS — APPROVED** [review]
Looks Perfect To Me

**Review by Luigi123 — APPROVED** [review]

**Review by timbuckley — APPROVED** [review]
lgtm, should improve DX greatly.

**Review by denniscampos — APPROVED** [review]

**Comment by ryanyogan** [comment]
When your at work and ask if anyone can review your PR... and everyone is too busy reviewing this 🤣🤣🚀

**Review by jclusso — CHANGES_REQUESTED** [review]
Seems bloated.

**Review by AZagatti — APPROVED** [review]
LGTM

**Review by park012241 — APPROVED** [review]

**Review by Claudemirovsky — APPROVED** [review]
LGTM

**Review by janfranco27 — APPROVED** [review]

**Comment by janklimo** [comment]
Ship it 🚢

**Comment by eldyvoon** [comment]
Lgtm, pls merge

**Review by bvoo — APPROVED** [review]

**Review by eldyvoon — APPROVED** [review]

**Comment by hansderly** [comment]
LGTM

**Review by kayode0x — APPROVED** [review]

**Comment by chrissnyder** [comment]
this pr already has been discussed 3x as long as #971. too much discussion, should just merge

**Comment by jdforsythe** [comment]
Burn. LGTM

**Comment by jpike88** [comment]
This PR is taking too long to merge, something as simple and inconsequential as this should have been merged in a couple of hours, tops!

**Review by anantsimran — APPROVED** [review]
For 🍿

**Comment by junaga** [comment]
@t3dotgg you know that `jsconfig.json` > `tsconfig.json` yet you make a show. is this why no one respects you? take a look at @fireship, he's a responsible creator.

**Review by airowe — APPROVED** [review]

**Review by narze — APPROVED** [review]
Quite surprising that #971 has no JSDocs in it, that's really bad for DX

**Review by SwooshyCueb — APPROVED** [review]

**Review by thammarith — APPROVED** [review]
LGTM

**Review by vsylvestre — APPROVED** [review]

**Review by NexonTreeHouse — APPROVED** [review]

**Comment by Greek** [comment]
lgtm

**Review by Greek — APPROVED** [review]

**Review by ven — APPROVED** [review]

**Comment by joshmanders** [comment]
Remember, this guys only contribution to anything of substance was putting together a shoddy next.js boilerplate using trpc's readme, then actual competent devs came in and built a real DX around it while he throws himself into as much drama and "Don't forget about wittle me in your debate on the merits of JS vs TS" to stroke his own narcissistic ego and to promote his shitty YouTube videos where he struggles to draw diagrams on Excalidraw for 45 minutes while he talks about scaling his 17 user "startup" app.

Theo, do us all a favor and leave real devs alone to do their thing. You had ZERO issues with Rich Harris removing TypeScript from Svelte, but now that 37Signals did it here, it's an issue? Double standard much?

**Review by pythongiant — APPROVED** [review]
LGTM 👍

**Review by apiel51 — APPROVED** [review]

**Review by superhooman — APPROVED** [review]

**Comment by TechStudent10** [comment]
LGTM of course Theo is on it haha

**Comment by ThePrimeagen** [comment]
Not gonna be active on Github reviewing typescript PRs tonight. I’m meeting a girl (a real one) in half an hour (wouldn’t expect a lot of you to understand anyway) so please don’t DM me asking me where I am (im with the girl, ok) you’ll most likely get aired because ill be with the girl (again I don’t expect you to understand) shes actually really interested in me and its not a situation i can pass up for some meaningless Github JavaScript degenerates (because ill be meeting a girl, not that you really are going to understand) this is my life now. Meeting women and not wasting my precious time online, I have to move on from such simple things and branch out (you wouldnt understand)

**Review by tyhuber1 — APPROVED** [review]

**Pull request closed without merging**
